### PR TITLE
Enhanced relay switch & selected channel memory for each relay

### PR DESCRIPTION
**Pull request by bob2402**

### CODE REVIEW

File: app/UI/nav.tsx [hunk 1]
```diff
@@ -80,11 +81,29 @@ export class NavBar extends Component<Props, State> {
...
+                return 4;
+            default:
+                return 0;
+        }
+    };
+
     state: State = {
         selectedRelay: "",
-        activeIndex: 0,
+        activeIndex: this.initialActiveIndex(),
```
Comment (BlowaterNostr) [inline app/UI/nav.tsx:103]: maybe `activeIndex` itself is a bad idea. Always prefer to string over number for indexing.

File: app/UI/relay-switch-list.tsx [hunk 2]
```diff
@@ -125,6 +150,9 @@ export async function getRelayInformation(url: string) {
         }
 
         const detail: RelayInformation = await res.json();
+        if (!detail.icon) {
+            detail.icon = httpURL + "favicon.ico";
```
Comment (BlowaterNostr) [inline app/UI/relay-switch-list.tsx:154]: use `path.join`

File: app/UI/search_model.ts [hunk 3]
```diff
@@ -10,5 +10,6 @@ export type SelectConversation = {
 };
 export type SelectChannel = {
     type: "SelectChannel";
-    name: string;
+    relay: string;
```
Comment (BlowaterNostr) [inline app/UI/search_model.ts:13]: current selected relay is a global state, we don't need to specify it here.

File: app/UI/nav.tsx [hunk 4]
```diff
@@ -47,6 +47,7 @@ type Props = {
     emit: emitFunc<NavigationUpdate | SelectRelay>;
     installPrompt: InstallPrompt;
     pool: ConnectionPool;
+    activeNav: string;
```
Comment (BlowaterNostr) [inline app/UI/nav.tsx:50]: No need

**Review by BlowaterNostr — CHANGES_REQUESTED** [review]

File: app/UI/nav.tsx (hunk 1 above)
Comment (bob2402) [reply app/UI/nav.tsx]: Okay, I replaced it with `activeNav`. All taken from the model(global state).

File: app/UI/nav.tsx (hunk 4 above)
Comment (bob2402) [reply app/UI/nav.tsx]: Need taken from the model(global state)

**Pull request merged**